src/State/SerializerAwareProviderTrait.php

```diff
@@ -30,6 +30,13 @@ trait SerializerAwareProviderTrait
 
     public function setSerializerLocator(ContainerInterface $serializerLocator): void
     {
+        trigger_deprecation(
+            'api-platform/core',
```

Comment:
Isn't it supposed to be `api-platform/state` component instead?

```suggestion
            'api-platform/state',
```